### ISSUE TITLE
[json-path] Indexed access on missing properties is confusing

### ISSUE THREAD

**Issue by wormuth** [rest-assured/rest-assured#1251]
**TL;DR**
Indexed access to non-existing properties should deliver `null` consistently instead of throwing an ``IllegalArgumentException`` under certain circumstances.

**Decription, Background**
I am trying to retrieve data from a rather dynamic JSON object via Json Path:
```javascript
{
  "items": [{
      "a": [
        "item1_a_v1",
        "item1_a_v2"
      ],
      "b": [
        "item1_b_v1",
        "item1_b_v2"
      ]
    }, {
      "a": [
        "item2_a_v1",
        "item2_a_v2"
      ]
    }, {
      "c": [{
          "data": [
            "item3_c_v1",
            "item3_c_v2"
          ]
        }
      ]
    }
  ]
}
```
(I did not make up this model myself, basically its the akeneo web api model I'm trying to process and the example given above is the result of me trying to boil it down to be able to demonstrate the problem).

It is "dynamic" in the sense that I do not know which properties an item has:
the list of possible properties is defined and finite (a, b, c), but which of these are present for a particular item may vary - just like shown above.

At the end of the day this "dynamic" model with me trying to access potentially missing arrays in an indexed kind of way revealed the problem I'm trying to demonstrate in this issue.

```java
// b is missing, so null is delivered - passing as expected:
assertThat( new JsonPath( JSON ).get( "items[1].b" ), nullValue() );
// b is missing, so trying to access b.c should deliver null - passing as expected:
assertThat( new JsonPath( JSON ).get( "items[1].b.c" ), nullValue() );
// Unexcepted:
// indexed access on b fails with an IllegalArgumentException instead of delivering null:
assertThat( new JsonPath( JSON ).get( "items[1].b[0]" ), nullValue() );
// Same (unexpected) IllegalArgumentException here:
assertThat( new JsonPath( JSON ).get( "items[2].d[0].data[0]" ), nullValue() );

// Even more unexpected and particularly lucky for me: NO Exception here, but null as I hoped - even though d does not exist:
assertThat( new JsonPath( JSON ).get( "items[2].d.data[0][0]" ), nullValue() );
// yet this trick does NOT work for the "simple" property b:
assertThat( new JsonPath( JSON ).get( "items.b[1][0]" ), nullValue() ); // still throws IllegalArgumentException
```
By accessing the indexed properties like this
``items[2].optional_prop.data[0][0]``
as opposed to 
``items[2].optional_prop[0].data[0]``
I got my particular data read, as it delivers `null`, whether `optional_prop` exists or not.

Yet this only works for "nested" properties like `c`, not in "simple" cases like `b`.

I have created a unit test to demonstrate the issue, see https://github.com/wormuth/rest-assured/blob/master/json-path/src/test/java/io/restassured/path/json/JsonPathIndexedAccessOnMissingPropertiesTest.java

**Comment by wormuth** [comment]
Reproducible with `5.0.1`, obviously fixed with `5.2.0`
